src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -205,136 +208,131 @@ function PanelManager(props) {
 
             </Paper>
         )
+    };
 
-};
-// const editRecord = (id) => (e) => {
-//     console.log("editRecord()")
-//     console.log(id);
-//     e.preventDefault();
+    // const editRecord = (id) => (e) => {
 
-//     let editingRecord = recordList.find((record) => {return record.id === id});
-//     let recordToUpload = {};
+    //     e.preventDefault();
 
-//     if(editingRecord) {
-//         Object.keys(editingRecord).forEach((key) => {
-//             recordToUpload[key] = editingRecord[key]
-//         });
+    //     let editingRecord = recordList.find((record) => {return record.id === id});
+    //     let recordToUpload = {};
 
-        
-//         props.updateFunction(id, recordToUpload);
-//     }
-//     editingRecord.editing = !editingRecord.editing;
-//     let updatedRecordList = recordList.map((record) => {
-//         if(record.id == id){
-//             return editingRecord;
-//         } else {
-//             return record;
-//         }
-//     });
-//     setRecordList(updatedRecordList);
-// };
-
-const editRecord = (index) => (e) => {
-    console.log("editRecord()")
-    console.log(index);
-    e.preventDefault();
-
-    // let editingRecord = recordList.find((record) => {return record.id === id});
-    let recordToUpload = {};
-
-    if(editingRecord) {
-        Object.keys(editingRecord).forEach((key) => {
-            recordToUpload[key] = editingRecord[key]
+    //     if(editingRecord) {
+    //         Object.keys(editingRecord).forEach((key) => {
+    //             recordToUpload[key] = editingRecord[key]
+    //         });
+    //         props.updateFunction(id, recordToUpload);
+    //     }
+    //     editingRecord.editing = !editingRecord.editing;
+    //     let updatedRecordList = recordList.map((record) => {
+    //         if(record.id == id){
+    //             return editingRecord;
+    //         } else {
+    //             return record;
+    //         }
+    //     });
+    //     setDirectoryData(updatedRecordList);
+    // };
+
+    const editRecord = (index) => (e) => {
+        e.preventDefault();
+        const recordToEdit = {};
+
+        let mutableDirectoryData = directoryData;
+        Object.keys(directoryData).map(key => {
+            if (key == "editing") {
+                // Do nothing
+            } else if (key == "ids") {
+                recordToEdit["id"] = directoryData["ids"][index]
+
+            } else {
+                recordToEdit[key] = directoryData[key]["values"][index];
+            }
+  
         });
-        props.updateFunction(id, recordToUpload);
+        props.updateFunction(recordToEdit["id"], recordToEdit);
+
+        mutableDirectoryData["editing"][index] = !directoryData["editing"][index];
+        setDirectoryData(mutableDirectoryData)
+
```

Comment:
Here, the correct value shows true. However, the end of this function causes a re-render, which sets everything to false. The editRecordRow never renders...the conditional on line 360 never returns true.